src/main/java/executor/service/service/parallelflowexecutor/impls/subscribers/DefaultPairGeneratorService.java

```diff
@@ -0,0 +1,78 @@
+package executor.service.service.parallelflowexecutor.impls.subscribers;
+
+import executor.service.model.proxy.ProxyConfigHolder;
+import executor.service.model.scenario.Scenario;
+import executor.service.queue.ProxyQueue;
+import executor.service.queue.ScenarioQueue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DefaultPairGeneratorService implements PairGeneratorService {
+    private final ProxyQueue proxyQueue;
+    private final ScenarioQueue scenarioQueue;
+
+    private final ProxyConfigHolder defaultProxy;
+
+    public DefaultPairGeneratorService(ProxyQueue proxyQueue, ScenarioQueue scenarioQueue, ProxyConfigHolder defaultProxy) {
+        this.proxyQueue = proxyQueue;
+        this.scenarioQueue = scenarioQueue;
+        this.defaultProxy = defaultProxy;
+    }
+
+
+    @Override
+    public List<Pair> generatePairs() {
+        List<Pair> pairs = new ArrayList<>();
+        List<ProxyConfigHolder> proxies;
+        List<Scenario> scenarios;
+        try {
+            proxies = proxyQueue.getAllProxy();
+            scenarios = scenarioQueue.getAllScenario();
+            if (!scenarios.isEmpty() && !proxies.isEmpty()) {
+                if (proxies.size() == scenarios.size()) {
+                    pairs = createOneToOnePairs(proxies, scenarios);
+                } else {
+                    pairs = createMixPairs(proxies, scenarios);
+                }
+            } else if (!scenarios.isEmpty()) {
+                proxies.add(defaultProxy);
+                pairs = createMixPairs(proxies, scenarios);
+            }
```

Comment:
There are four main cases:
1. scenarios and proxy has the same size
2. scenarios and proxy has different size, and both of them more than zero
3. there are no proxy
4. there are no scenarios

you describe three cases, but what will happen with proxy if there no scenarios?